libraries/eosiolib/core/eosio/crypto_ext.hpp

```diff
@@ -48,14 +49,182 @@ namespace eosio {
          return dg;
       }
    }
-      
+
    /**
     *  @defgroup crypto Crypto
     *  @ingroup core
     *  @brief Defines API for calculating and checking hashes which
     *  require activating crypto protocol feature
     */
 
+   /**
+    * Abstracts G1 and G2 points. Ensures sizes of x and y are valid
+    *
+    *  @ingroup crypto
+    */
+   struct point_view {
+      /**
+       * Pointer to the x coordinate
+       */
+      char* x;
+
+      /**
+       * Pointer to the y coordinate
+       */
+      char* y;
+
+      /**
+       * Number of bytes in each of x and y
+       */
+      uint32_t size;
+
+      /**
+       * Construct a point given x and y
+       *
+       * @param x_ - The x coordinate, a vector of chars
+       * @param y_ - The y coordinate, a vector of chars
+       */
+      point_view(std::vector<char>& x_, std::vector<char>& y_)
+      :x(x_.data()), y(y_.data()), size(x_.size())
+      {
+         eosio::check( x_.size() == y_.size(), "x's size must be equal to y's" );
+      };
+
+      /**
+       * Construct a point given a serialized point
+       *
+       * @param p - The serialized point
+       */
+      point_view(std::vector<char>& p)
+      :x(p.data()), y(p.data() + p.size()/2), size(p.size()/2)
+      {
+      };
+
+      /**
+       *  Returns packed x and y
+       */
+      std::vector<char> packed() const {
```

Comment:
Unless there are some recent changes that I'm not aware of, we are using non-class member functions like:
`eosio::pack` and `eosio::unpack` all over the code. So I suggest to adhere to those standards.

Reply:
The name here is misleading. The intention was to provide a way to serialize, as a complimentary to `point_view(std::vector<char>& p)`. Change the name to `serialized` and remove `unpack` which is a duplicate of `point_view(std::vector<char>& p)`.

Reply:
ok, got you. Anyway it looks like some non-standard approach for serialization. There is `CDT_REFLECT` macro in the `eosiolib`, have you considered it? it looks like a good approach for your purpose. it could be a bit tricky to use in your case but that way we can stick to standard approach. please let me know what do you think.